app/services/imports/lettings_logs_field_import_service.rb

```diff
@@ -453,5 +455,28 @@ def update_old_form_id(xml_doc)
         @logger.warn("lettings log with old id #{old_id} not found")
       end
     end
+
+    def update_postcode_full(xml_doc)
+      return if meta_field_value(xml_doc, "form-name").include?("Sales")
+
+      old_id = meta_field_value(xml_doc, "document-id")
+      record = LettingsLog.find_by(old_id:)
+      return @logger.info("lettings log #{record.id} is not from 2022/23 collection period, skipping update") if record.collection_start_year != 2022
+      return @logger.warn("Could not find record matching legacy ID #{old_id}") if record.blank?
+      return @logger.info("lettings log #{record.id} has a value for postcode_full, skipping update") if record.postcode_full.present?
+
+      record.postcode_full = compose_postcode(xml_doc, "POSTCODE", "POSTCOD2")
```

Comment:
do we also have this issue for previous postcodes (PPOSTC1 and PPOSTC2)?

Reply:
same as above